### DISCUSSION TITLE
Finetuning LLM models with PDF documents in h2o-llmstudio

### DISCUSSION THREAD

**Discussion by sunilswain** [h2oai/h2o-llmstudio#719] [Q&A] [answered]
Hi, 
does anyone know how we can use pdf documents(E.g - Travel Policy document) and use them to finetune our model using llmstudio.

Thanks.

**Comment by psinger** [comment, marked as answer]
Hi,

you would first need to generate input/output pairs for your documents. I answered a similar question already here that might be helpful:
https://github.com/h2oai/h2o-llmstudio/issues/522

If you want to just to next token training on the text of your pdfs, then you would need to transform it to a csv file with raw text and follow this: https://docs.h2o.ai/h2o-llmstudio/faqs#what-if-my-data-is-not-in-question-and-answer-form-and-i-just-have-documents-how-can-i-fine-tune-the-llm-model
